core/main/src/service/apps/apps_updater.rs

```diff
@@ -39,16 +144,46 @@ impl AppsUpdater {
         client: ExtnClient,
         ignore_list: Vec<String>,
         uninstalls_enabled: bool,
+        sideload_support: bool,
+        failed_app_installs: Vec<FailedAppInstall>,
+        failed_app_installs_persist_path: String,
     ) -> AppsUpdater {
         AppsUpdater {
             state: AppsUpdaterState {
                 client,
                 ignore_list,
                 uninstalls_enabled,
+                sideload_support,
+                failed_app_installs: Arc::new(RwLock::new(failed_app_installs)),
+                failed_app_installs_persist_path,
+                pending_installs: Arc::new(RwLock::new(vec![])),
             },
             streamer: DefaultExtnStreamer::new(),
         }
     }
+
+    pub async fn init(state: &PlatformState) -> AppsUpdater {
+        let ignore_list = vec![state.get_device_manifest().applications.defaults.main];
+        let mut extn_client = state.get_client().get_extn_client();
+        let feats = state.get_device_manifest().get_features();
+        let sideload_support = DeviceInfo::is_debug(&mut extn_client).await;
```

Comment:
Most of this method was moved out of main_context_processor to here.
I have added a check to query the device is a debug device before allowing the sideload logic.